src/components/ValidateCodeActionModal/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -0,0 +1,257 @@
+    validatePendingAction?: PendingAction | null;
+
+    /** The error of submitting  */
+    validateError?: Errors;
+
+    /** Function is called when submitting form  */
+    handleSubmitForm: (validateCode: string) => void;
+
+    /** Function to clear error of the form */
+    clearError: () => void;
+};
+
+type BaseValidateCodeFormProps = BaseValidateCodeFormOnyxProps & ValidateCodeFormProps;
+
+function BaseValidateCodeForm({
+    account = {},
+    hasMagicCodeBeenSent,
+    autoComplete = 'one-time-code',
+    innerRef = () => {},
+    validateCodeAction,
+    validatePendingAction,
+    validateError,
+    handleSubmitForm,
+    clearError,
+}: BaseValidateCodeFormProps) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const [formError, setFormError] = useState<ValidateCodeFormError>({});
+    const [validateCode, setValidateCode] = useState('');
+    const inputValidateCodeRef = useRef<MagicCodeInputHandle>(null);
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing -- nullish coalescing doesn't achieve the same result in this case
+    const shouldDisableResendValidateCode = !!isOffline || account?.isLoading;
+    const focusTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    useImperativeHandle(innerRef, () => ({
+        focus() {
+            inputValidateCodeRef.current?.focus();
+        },
+        focusLastSelected() {
+            if (!inputValidateCodeRef.current) {
+                return;
+            }
+            if (focusTimeoutRef.current) {
+                clearTimeout(focusTimeoutRef.current);
+            }
+            focusTimeoutRef.current = setTimeout(() => {
+                inputValidateCodeRef.current?.focusLastSelected();
+            }, CONST.ANIMATED_TRANSITION);
+        },
+    }));
+
+    useFocusEffect(
+        useCallback(() => {
+            if (!inputValidateCodeRef.current) {
+                return;
+            }
+            if (focusTimeoutRef.current) {
+                clearTimeout(focusTimeoutRef.current);
+            }
+            focusTimeoutRef.current = setTimeout(() => {
+                inputValidateCodeRef.current?.focusLastSelected();
+            }, CONST.ANIMATED_TRANSITION);
+            return () => {
+                if (!focusTimeoutRef.current) {
+                    return;
+                }
+                clearTimeout(focusTimeoutRef.current);
+            };
+        }, []),
+    );
+
+    useEffect(() => {
+        if (!validateError) {
+            return;
+        }
+        clearError();
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, [clearError, validateError]);
+
+    useEffect(() => {
+        if (!hasMagicCodeBeenSent) {
+            return;
+        }
+        inputValidateCodeRef.current?.clear();
```

Comment:
{BZ CHECKLIST} inputValidateCodeRef.current?.clear() was set too early, during the RHP animation. which caused this minor transition flickering issue https://github.com/Expensify/App/issues/73030